### ISSUE TITLE
IDISP016 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#160]
```cs
namespace Valid
{
    using System;
    using System.Collections.Generic;

    sealed class C : IDisposable
    {
        private readonly List<(IDisposable, IDisposable)> xs = new List<(IDisposable, IDisposable)>();

        public void Dispose()
        {
            foreach (var tuple in this.xs)
            {
                tuple.Item1.Dispose();
                tuple.Item2.Dispose();
            }
        }
    }
}
```
